types/node/index.d.ts

```diff
@@ -5831,3 +5832,818 @@ declare module "async_hooks" {
      */
     export function createHook(options: HookCallbacks): AsyncHook;
 }
+
+declare module "http2" {
+    import * as events from "events";
+    import * as fs from "fs";
+    import * as http from "http";
+    import * as net from "net";
+    import * as stream from "stream";
+    import * as tls from "tls";
+    import * as url from "url";
+
+    export interface IncomingHttpHeaders extends http.IncomingHttpHeaders {}
+    export interface OutgoingHttpHeaders extends http.OutgoingHttpHeaders {}
+
+    // Http2Stream
+
+    export interface StreamPriorityOptions {
+        exclusive?: boolean;
+        parent?: number;
+        weight?: number;
+        silent?: boolean;
+    }
+
+    export interface StreamState {
+        localWindowSize?: number;
+        state?: number;
+        streamLocalClose?: number;
+        streamRemoteClose?: number;
+        sumDependencyWeight?: number;
+        weight?: number;
+    }
+
+    export interface ServerStreamResponseOptions {
+        endStream?: boolean;
+        getTrailers?: (trailers: IncomingHttpHeaders) => void;
+    }
+
+    export interface StatOptions {
+        offset: number;
+        length: number;
+    }
+
+    export interface ServerStreamFileResponseOptions {
+        statCheck?: (stats: fs.Stats, headers: IncomingHttpHeaders, statOptions: StatOptions) => void;
+        getTrailers?: (trailers: IncomingHttpHeaders) => void;
+        offset?: number;
+        length?: number;
+    }
+
+    export interface Http2Stream extends stream.Duplex {
+        readonly aborted: boolean;
+        readonly destroyed: boolean;
+        priority(options: StreamPriorityOptions): void;
+        readonly rstCode: number;
+        rstStream(code: number): void;
+        rstWithNoError(): void;
+        rstWithProtocolError(): void;
+        rstWithCancel(): void;
+        rstWithRefuse(): void;
+        rstWithInternalError(): void;
+        readonly session: Http2Session;
+        setTimeout(msecs: number, callback?: () => void): void;
+        readonly state: StreamState;
+
+        addListener(event: string, listener: (...args: any[]) => void): this;
```

Comment:
I think the most general overload should go *last*, otherwise none of the previous later overloads will be reached.